src/elements/content-sidebar/activity-feed/task-new/TaskStatus.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import * as React from 'react';
+import { FormattedMessage } from 'react-intl';
+import IconCheck from '../../../../icons/general/IconCheck';
+import IconReject from '../../../../icons/general/IconClose';
+import { TASK_NEW_APPROVED, TASK_NEW_REJECTED, TASK_NEW_COMPLETED, TASK_NEW_INCOMPLETE } from '../../../../constants';
+import messages from '../../../common/messages';
+
+type Props = {|
+    status: TaskStatus,
+|};
+
+const ICON_SIZE = 14;
+
+const IconIncompleteNoSvg = props => <span {...props}>?</span>;
+
+const StatusIcon = ({ status, ...rest }: { status: TaskStatus }) => {
+    switch (status) {
+        case TASK_NEW_APPROVED:
+        case TASK_NEW_COMPLETED:
+            return <IconCheck width={ICON_SIZE} height={ICON_SIZE} {...rest} />;
+        case TASK_NEW_REJECTED:
+            return <IconReject width={ICON_SIZE} height={ICON_SIZE} {...rest} />;
+        case TASK_NEW_INCOMPLETE:
+            return <IconIncompleteNoSvg width={ICON_SIZE} height={ICON_SIZE} {...rest} />;
+        default:
+            return null;
+    }
+};
+
+const StatusMessage = ({ status }: { status: TaskStatus }) => {
```

Comment:
We usually try to stick to one component per file.

Reply:
https://github.com/airbnb/javascript/pull/2006

Reply:
It seems like not exporting the components would make it difficult to test them in isolation. @priyajeet, thoughts?